### ISSUE TITLE
Could not validate signature

### ISSUE THREAD

**Issue by realsando** [orbitdb/orbitdb#868]
We are trying to test our application with OrbitDB.
Everything seems to be running fine, but after a bit of load testing with different use case we often run in the following error, which seems to be happening randomly :

```
 UnhandledPromiseRejectionWarning: Error: Could not validate signature "304...." for entry "zdpu...." and key "0440..."
    at verify (E:\Blockchain\dbnode\node_modules\ipfs-log\src\log.js:414:27)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at runNextTicks (internal/process/task_queues.js:62:3)
    at processImmediate (internal/timers.js:434:9)
    at async pMap.concurrency (E:\Blockchain\dbnode\node_modules\ipfs-log\src\log.js:420:7)
    at async E:\Blockchain\dbnode\node_modules\p-map\index.js:57:22
(Use `node --trace-warnings ...` to show where the warning was created)

```
It seems to happen quite randomly, so I haven't figured out the clear way to reproduce this issue. This seems to be linked to internal signature verification mechanism within orbitdb. Any idea wherer this might come from and how to prevent it from happening ?

**Comment by sentryjs** [comment]
Could this be related? orbitdb/ipfs-log#242

**Comment by paskl** [comment]
interesting ! I did post has an object. After stringify-ing it the issue do not seem to re-occur so far. Thank you!

**Comment by sentryjs** [comment]
So after I posted my comment above I went to ask in the Gitter chat. @kowalski basically told me to use primitives and @aphelionz called this "The Case of the Non-Deterministic JSON String Encoding" as if it were a common question. In my implementation I do use objects, but if you have things like Date objects in them you would want to convert to the string representation. This solved my problem and I wanted to update you so you understand that basic objects aren't exactly the problem and stringify isn't exactly the solution you were looking for.

Hope this was clear, good luck!
